html/controls/pt/index.js

```diff
@@ -213,20 +213,23 @@ function submitPenalty() {
 	penaltyEditor.dialog('close');
 }
 
-var penaltyCodeRegex = /^PenaltyCode.([^\(]+)\(([^\)]+)\)/;
 function penaltyCode(k, v) {
-	match = k.match(penaltyCodeRegex);
-	if (match == null)
-		return;
-	var type = match[1];
-	var code = match[2];
+	var code = k.split('.').pop();
 
-	var div = $('.Codes .' + type + '[code="' + code + '"]');
-	if (div.length > 0) {
-		if (v == null) {
-			div.detach();
-			return;
+
+		if(code !== 'FO') {
```

Comment:
Maybe we should do the FO code handling entirely in the frontend? It feels odd to list it as a penalty code.